apps/src/netsim/NetSimSendWidget.js

```diff
@@ -54,24 +86,123 @@ NetSimSendWidget.createWithin = function (element, connection) {
 NetSimSendWidget.prototype.bindElements_ = function () {
   this.rootDiv_ = $('#netsim_send_widget');
   this.toAddressTextbox_ = this.rootDiv_.find('#to_address');
-  this.payloadTextbox_ = this.rootDiv_.find('#payload');
+  this.toAddressTextbox_.bind('keyPress', function (event) {
+    // Don't block control combinations (copy, paste, etc);
+    if (event.metaKey || event.ctrlKey) {
+      return true;
+    }
+
+    // Prevent non-digit characters
+    var charCode = !event.charCode ? event.which : event.charCode;
+    if (charCode >= 32 && charCode <= 126) {
```

Comment:
Would prefer to see these hard coded so that I know what 32 and 126 mean. I know we use a KeyCode structure elsewhere with some of these defined.